### PR TITLE
add code and test for us18 and test for main page

### PR DESCRIPTION
**Pull request by JimmyTrulock**
add code and testing for us18
add testing for merchant index home page

### CODE REVIEW

File: spec/features/merchants/invoices/show_spec.rb [hunk 1]
```diff
@@ -64,12 +64,26 @@
...
+      visit merchant_invoice_path(@merchant_1, @invoice_1)
+
+      expect(page).to have_content("Total revenue: $1,000.00")
 
-    expect(page).to have_content("Total revenue: $1,000.00")
+    end
+  end
 
+  #US 18
+  describe "Merchant Invoice Show Page: Update Item Status" do
```
Comment (chuckrenny) [inline spec/features/merchants/invoices/show_spec.rb:77]: Testing needs to show the current status is selected and options to select a new status to update the attribute. US: I see that each invoice item status is a select field
And I see that the invoice item's current status is selected
When I click this select field,
Then I can select a new status for the Item

**Comment by chuckrenny** [comment]
Throughout the files, we want to use advanced routing. Eg: merchants_path, admin_invoices_path, etc

File: app/controllers/merchants/invoices_controller.rb [hunk 2]
```diff
@@ -8,4 +8,15 @@ def show
...
+  def update
+    @item = Item.find(params[:item_id])
+    
+    if params[:status] == "Enabled"
+      @item.update(status: 0)
+    elsif params[:status] == "Disabled"
+      @item.update(status: 1)
+    end
+
+    redirect_to "/merchants/#{params[:merchant_id]}/invoices/#{params[:id]}"
```
Comment (chuckrenny) [inline app/controllers/merchants/invoices_controller.rb:20]: Advanced routing here

File: app/views/merchants/invoices/show.html.erb [hunk 3]
```diff
@@ -5,17 +5,21 @@
...
   <p>Quantity sold: <%= item.quantity %></p>
   <p>Price sold: <%= number_to_currency(item.price_total) %>
-  <p>Status: <% if item.shipping_status == 2 %>
-                <p>Shipped<P>
-              <% elsif item.shipping_status == 1%>
-                <p>Packaged</p>
-                <% else %>
-                <p>Pending</p>
-             <% end %>
+  <p>Status:  <%= form_with(url: "/merchants/#{@merchant.id}/invoices/#{@invoice.id}", method: :patch, local: true) do |form| %>
```
Comment (chuckrenny) [inline app/views/merchants/invoices/show.html.erb]: Routes here

File: spec/features/merchants/index_spec.rb [hunk 4]
```diff
@@ -7,6 +7,15 @@
   end
 
   describe "index page" do
+    it "shows all merchants and link to the merchant show" do
+      visit "/merchants"
```
Comment (chuckrenny) [inline spec/features/merchants/index_spec.rb:11]: this would be a visit merchants_path

File: spec/features/merchants/index_spec.rb [hunk 5]
```diff
@@ -7,6 +7,15 @@
...
+
+      expect(page).to have_content(@merchant_1.name)
+
+      click_link("#{@merchant_1.name}")
+
+      expect(current_path).to eq("/merchants/#{@merchant_1.id}/dashboard")
```
Comment (chuckrenny) [inline spec/features/merchants/index_spec.rb:17]: I believe this would be a merchant_dashboard_path(@merchant_1.id)

**Comment by kameronk92** [comment]
Nice team work to resolve the merge conflict on this one!

**Pull request merged**